src/components/transport_manager/CMakeLists.txt

```diff
@@ -53,6 +53,13 @@ target_sources(${PROJECT_NAME}
     "${CMAKE_CURRENT_LIST_DIR}/src/transport_adapter/transport_adapter_listener_impl.cc"
 )
 
+if(BUILD_TESTS)
+  target_sources(${PROJECT_NAME}
+    PRIVATE
+      "${CMAKE_CURRENT_LIST_DIR}/src/iap2_emulation/iap2_transport_adapter.cc"
```

Comment:
@SKobziev @BSolonenko  not clear,  why do we add additional sources within tests to the project?

Reply:
@LuxoftAKutsan We have `include` that depend on BUILD_TESTS
https://github.com/BSolonenko/sdl_core/blob/df9b67026d90f0f9cdf04bec7dd3c7666a25db87/src/components/transport_manager/src/transport_manager_default.cc#L47-L49